### ISSUE TITLE
Experimental discrepancy

### ISSUE THREAD

**Issue by shenyihang666** [saranya-venkatraman/gpt-who#3]
Through repeated experiments, it is found that the structure obtained is different from the experimental results in this paper. Take the training set and test set of gpt2-xl in TuringBench data set as an example：

![Image](https://github.com/user-attachments/assets/6791d125-ce1c-4cf0-a315-0014d9dbd75f)

And I found that the value of parameter N in the code is not 20 as mentioned in the paper.

Please answer my doubts by the author, my Google mailbox is shenchenli0215@gmail.com

**Comment by saranya-venkatraman** [comment]
@shenyihang666  We have found that there are possibly 3 different splits of the turingbench dataset (since the dataset was updated). Yours might be a different split than the one we used for our experiments, though, as pointed out, we do treat N as a hyperparameter and you should tune it for a new dataset/split. From wider experiments on newer datasets, we recommend a value of N at 50 or 60 tokens. Please feel free to follow up on email at szv4@psu.edu. Thanks!
